### PR TITLE
doc: update doc guidelines and building content

### PR DESCRIPTION
**Pull request by dbkinder**

### CODE REVIEW

**Review by tomlenth — APPROVED** [review]
That's a lot of solid updates... you've been busy!

File: developer-guides/doc_guidelines.rst [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
+The HTML content can then be viewed using a web browser. These ``.md`` and
+``.rst`` files are maintained in the project's GitHub repos and processed to
+create the `OPEA Project documentation`_ website.
 
 .. note:: While GitHub supports viewing `.md` and `.rst` content with your browser on the
    `github.com` site, markdown and reST extensions are not recognized there, so the
    best viewing experience is through the `OPEA Project documentation`_ github.io
-   website.
+   website. The github.io site also provides navigation and searching that makes
+   it easier for find and read what you're looking for.
```
Comment (mkbhanda) [inline developer-guides/doc_guidelines.rst:19]:
```suggestion
   it easier to find and read what you're looking for.
```

File: developer-guides/doc_guidelines.rst [hunk 2]
```diff
@@ -79,6 +80,9 @@ directive to point to other documents that may include additional ``toctree``
 directives of their own, ultimately collecting all the content into an
```
Comment (mkbhanda) [inline developer-guides/doc_guidelines.rst:80]: directives? or directories?

**Review by mkbhanda — APPROVED** [review]

**Comment by mkbhanda** [comment]
@dbkinder just resolve those two comments and we merge.

File: developer-guides/doc_guidelines.rst (hunk 2 above)
Comment (dbkinder) [reply developer-guides/doc_guidelines.rst:80]: toctree directives (it's a ReStructuredText term).

File: developer-guides/doc_guidelines.rst (hunk 1 above)
Comment (dbkinder) [reply developer-guides/doc_guidelines.rst]: You can have a second career as a grammar/spelling checker :)  Thanks.

**Pull request merged**